src/reyna/Preferences.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Reyna
+{
+    using System;
+    using Microsoft.Win32;
+
+    internal static class Preferences
+    {
+        private const string StorageSizeLimitKeyName = "StorageSizeLimit";
+
+        internal static long StorageSizeLimit
+        {
+            get
+            {
+                return GetRegistryValue(StorageSizeLimitKeyName, -1);
+            }
+        }
+
+        internal static int ForwardServiceTemporaryErrorBackout
+        {
+            get
+            {
+                return GetRegistryValue("TemporaryErrorBackout", 5 * 60 * 1000);
+            }
+        }
+
+        internal static int ForwardServiceMessageBackout
+        {
+            get
+            {
+                return GetRegistryValue("MessageBackout", 1000);
+            }
+        }
+
+        internal static TimeRange CellularDataBlackout
+        {
+            get
+            {
+                try
+                {
+                    int minuteOfDayFrom = GetRegistryValue("DataBlackou:From", -1);
+                    int minuteOfDayTo = GetRegistryValue("DataBlackout:To", -1);
+                    if (minuteOfDayFrom == -1 && minuteOfDayTo == -1)
```

Comment:
shouldn't it be || instead of &&?
if one of those variables is -1 and the other is not we will not get into the if statement, but throw an exception from the `Time` constructor. It's going to be caught by the `catch` clause, but we can just prevent that, right?
@ivanbokii